udf-examples/pom.xml

```diff
@@ -58,14 +51,8 @@
     </dependency>
     <dependency>
         <groupId>com.nvidia</groupId>
-        <!--
-        This should depend on rapids-4-spark_${scala.binary.version} instead, but that dependency
-        only exists after the package phase. External projects should depend on
-        rapids-4-spark_${scala.binary.version}.
-        -->
-        <artifactId>rapids-4-spark-sql_${scala.binary.version}</artifactId>
+        <artifactId>rapids-4-spark_${scala.binary.version}</artifactId>
```

Comment:
The original complaint when I tried to make this change earlier (see #3945) is that it breaks `mvn test` on a clean build because this artifact does not exist until the `package` phase is executed.  That's one, but not all, of the main motivations to move this module into a separate build.